tests/python_validation/python_validate.py

```diff
@@ -0,0 +1,205 @@
+#/usr/bin/env python2
+
+""" 
+
+python2 python_validate.py [-p] [-f] [file1 file2 file3]
+
+or 
+
+python3 python_validate.py [-p] [-f] [file1 file2 file3]
+
+-f  (--fixup)  Automatically attempt to fix obvious python3 errors using reindent and 2to3.
+-p  (--pep8)   Run flake8 to test adherence to pep8 coding stanaards.
+
+Script to attempt to validate python files and code snippets in the Fluidity code base against python2 and python3 syntax.
+
+"""
+
+from __future__ import print_function
+import sys
+import os
+import ast
+import subprocess
+import tempfile
+
+from lxml import etree
+
+import argparse
+
+def flake8_snippet(name, snippet):
+    """Run flake8 against the snippet passed in, with reporting name as 'name'."""
+    with tempfile.TemporaryFile(mode='w') as temp: 
+        temp.write(snippet)
+        temp.seek(0)
+        try:
+            subprocess.check_output(['flake8 --stdin-display-name="%s" -'%name],
+                                    stdin = temp, shell=True)
+        except subprocess.CalledProcessError as err:
+            print(err.output.decode("utf8"))
+
+def print_as_function(filename, add_future_import=False):
+    """Handrolled function to turn
+    print blah
+       into
+    print(blah)"""
+    lines = []
+
+    def sanitize(line, sep):
+        if len(sep)==0:
+            if 'print ' in line:
+                return line.replace('print ', 'print(')+')'
+            else:
+                return line
+        else:
+            return sep[0].join([sanitize(_, sep[1:]) for _ in line.split(sep[0])])
+
+    with open(filename, 'r') as temp:
+        for line in temp.readlines():
+            if len(line.strip())==0 or line.strip()[0]=='#':
+                pass
+            if 'print ' in line:
+                line = sanitize(line.rstrip('\n'), [';', '#'])+'\n'
```

Comment:
Why is `#` treated as a separator? Surely if we ignore corner cases of `#` being inside a string (which is fair enough), then we might as well cut off at the `#` ? Also parsing for `#` should happen before `;`